### ISSUE TITLE
Support Python without SSL (2)

### ISSUE THREAD

**Issue by vbraun** [shazow/urllib3#584]
This is essentially the same as issue #320, urllib3 fails to import if there is no ssl module (Tested on Python 2.7.9). There are various reasons one might not have it, e.g. self-complied Python without ssl headers, the steady stream of security vulnerabilities, or objections to the not-really-free openssl license. But right now pip requires ssl because it uses urllib3.
~~~~
In [1]: import urllib3
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-dbe53ea169b1> in <module>()
----> 1 import urllib3

/home/vbraun/Code/sage/local/lib/python2.7/site-packages/urllib3/__init__.py in <module>()
      8 
      9 
---> 10 from .connectionpool import (
     11     HTTPConnectionPool,
     12     HTTPSConnectionPool,

/home/vbraun/Code/sage/local/lib/python2.7/site-packages/urllib3/connectionpool.py in <module>()
     29 from .packages.ssl_match_hostname import CertificateError
     30 from .packages import six
---> 31 from .connection import (
     32     port_by_scheme,
     33     DummyConnection,

/home/vbraun/Code/sage/local/lib/python2.7/site-packages/urllib3/connection.py in <module>()
     43 from .packages.ssl_match_hostname import match_hostname
     44 
---> 45 from .util.ssl_ import (
     46     resolve_cert_reqs,
     47     resolve_ssl_version,

/home/vbraun/Code/sage/local/lib/python2.7/site-packages/urllib3/util/__init__.py in <module>()
      3 from .request import make_headers
      4 from .response import is_fp_closed
----> 5 from .ssl_ import (
      6     SSLContext,
      7     HAS_SNI,

/home/vbraun/Code/sage/local/lib/python2.7/site-packages/urllib3/util/ssl_.py in <module>()
     10 
     11 import errno
---> 12 import ssl
     13 
     14 try:  # Test for SSL features

/home/vbraun/Code/sage/local/lib/python/ssl.py in <module>()
     58 import textwrap
     59 
---> 60 import _ssl             # if we can't import it, let the error propagate
     61 
     62 from _ssl import OPENSSL_VERSION_NUMBER, OPENSSL_VERSION_INFO, OPENSSL_VERSION

ImportError: No module named _ssl
~~~~

**Comment by Lukasa** [comment]
Yeah, per #320 we should probably fix this up to behave in the way @dstufft originally expressed (just fail if you try to do anything with SSL).

**Comment by shazow** [comment]
I think the plan was that once we make sure the "insecure request" warning was confirmed to be properly triggering in all the right scenarios*, we could switch it into a warning exception which would cover this case.

Does that sound sensible?

* Seems that it's misfiring? in the AppEngine scenario (#583), need to figure out what we want to do here.

**Comment by Lukasa** [comment]
@shazow Sadly it doesn't work in that case. The problem is that importing urllib3 is entirely blocked.

**Comment by shazow** [comment]
Oh I thought we fixed that. That's not great.

**Comment by vbraun** [comment]
I created PR #586 to provide a way to test without ssl.

**Comment by shazow** [comment]
Fixed in #587
